unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,334 @@
+    def test_execute_batch_commands_batched_command_published(aggregator):
+        aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+
+        batched_command_input = {"type": "BATCHED", "transaction_id": TEST_TRANSACTION_ID,
+                                 "aggregator_uuid": TEST_AGGREGATOR_UUID,
+                                 "batch_commands": TEST_BATCH_COMMAND_DICT}
+        batched_channel_input = f"external//aggregator/{TEST_AGGREGATOR_UUID}/batch_commands"
+
+        aggregator.execute_batch_commands(is_blocking=False)
+        aggregator.redis_db.publish.assert_called_with(batched_channel_input,
+                                                       json.dumps(batched_command_input))
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_execute_batch_commands_api_time_out_raise_exception(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking",
+                   side_effect=AssertionError):
+            aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+            aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+            with pytest.raises(RedisAggregatorAPIException):
+                aggregator.execute_batch_commands()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    @pytest.mark.parametrize("is_blocking, trans_id_resp_buffer, expected_ret_val",
+                             [(True, {TEST_TRANSACTION_ID: TEST_RESPONSE}, TEST_RESPONSE),
+                              (False, {TEST_TRANSACTION_ID: TEST_RESPONSE}, None),
+                              (True, {}, None)])
+    def test_execute_batch_commands_returns_expected(aggregator, is_blocking,
+                                                     trans_id_resp_buffer, expected_ret_val):
+        aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+        aggregator._transaction_id_response_buffer = trans_id_resp_buffer
+
+        assert aggregator.execute_batch_commands(is_blocking=is_blocking) is expected_ret_val
+
+
+@pytest.mark.usefixtures("mock_grid_fee_calculation")
+class TestRedisAggregatorCalculateGridFee:
+    """Test cases for RedisAggregator's calculate_grid_fee method."""
+
+    @staticmethod
+    def test_calculate_grid_fee_called_with_args(aggregator):
+        aggregator.calculate_grid_fee("start_market",
+                                      "target_market",
+                                      "current_market_fee")
+        aggregator.grid_fee_calculation.calculate_grid_fee.assert_called_with(
+            "start_market",
+            "target_market",
+            "current_market_fee")
+
+    @staticmethod
+    def test_calculate_grid_fee_returns_expected(aggregator):
```

Comment:
Actually, you are testing whether the `aggregator.calculate_grid_fee` calls the `aggregator.grid_fee_calculation.calculate_grid_fee`, you are not testing whether the the grid_fee value is valid, right? SO I vote for renaming this method.

Reply:
Actually, the assertion regarding wether it was called or not is within the test above. 

I see now that you point this out, that this test might not be needed, because if the GridFeeCalculation instance works properly (and therefore its method calculate_grid_fee does as well), then this test should always pass. So this is somehow redundant. 
I believe now that the right thing to do here is get rid of this test.